R/write_results.R

```diff
@@ -1,127 +1,40 @@
-                                      annual_pd_changes, overall_pd_changes,
-                                      asset_type, calculation_level,
-                                      sensitivity_analysis_vars, iter_var,
+write_stress_test_results <- function(results_list, asset_type, iter_var,
                                       output_path) {
+  results_list$stress_test_results_comp %>%
+    readr::write_csv(file.path(
+      output_path,
+      glue::glue("stress_test_results_{asset_type}_comp_{iter_var}.csv")
+    ))
 
-  sensitivity_analysis_vars <- paste0(sensitivity_analysis_vars, "_arg")
-
-  results %>%
-    write_results_new(
-      path_to_results = output_path,
-      asset_type = asset_type,
-      level = calculation_level,
-      file_type = "csv",
-      sensitivity_analysis_vars = sensitivity_analysis_vars,
-      iter_var = iter_var
-    )
-
-  expected_loss <- expected_loss %>%
-    dplyr::select(
-      .data$scenario_name, .data$scenario_geography, .data$investor_name,
-      .data$portfolio_name, .data$company_name, .data$ald_sector,
-      .data$pd, .data$PD_change, .data$lgd, .data$exposure_at_default,
-      .data$expected_loss_baseline, .data$expected_loss_late_sudden,
-      !!!rlang::syms(sensitivity_analysis_vars)
-    ) %>%
-    dplyr::arrange(
-      .data$scenario_geography, .data$scenario_name, .data$investor_name,
-      .data$portfolio_name, .data$company_name, .data$ald_sector
-    )
-
-  expected_loss %>%
-    report_missings(
-      name_data = "Expected loss"
-    ) %>%
-    report_all_duplicate_kinds(
-      composite_unique_cols = c(
-        "scenario_name", "scenario_geography", "investor_name", "portfolio_name",
-        "company_name", "ald_sector",
-        sensitivity_analysis_vars
-      )
-    ) %>%
+  results_list$stress_test_results_port %>%
     readr::write_csv(file.path(
       output_path,
-      paste0("stress_test_results_", asset_type, "_comp_el_", iter_var,".csv")
+      glue::glue("stress_test_results_{asset_type}_port_{iter_var}.csv")
     ))
 
-  annual_pd_changes_sector <- annual_pd_changes %>%
-    dplyr::group_by(
-      .data$scenario_name, .data$scenario_geography, .data$investor_name,
-      .data$portfolio_name, .data$ald_sector, .data$year,
-      !!!rlang::syms(sensitivity_analysis_vars)
-    ) %>%
-    dplyr::summarise(
-      # ADO 2312 - weight the PD change by baseline equity because this represents the original exposure better
-      PD_change = weighted.mean(x = .data$PD_change, w = .data$equity_t_baseline, na.rm = TRUE),
-      .groups = "drop"
-    ) %>%
-    dplyr::ungroup() %>%
-    dplyr::arrange(
-      .data$scenario_geography, .data$scenario_name, .data$investor_name,
-      .data$portfolio_name, .data$ald_sector, .data$year
-    )
+  results_list$stress_test_results_comp_el %>%
+    readr::write_csv(file.path(
+      output_path,
+      paste0("stress_test_results_", asset_type, "_comp_el_", iter_var, ".csv")
```

Comment:
why not always use `glue::glue()` for consistency?

Reply:
done